File: code-sec-evaluator/backend/app/schemas/worker.py
"""角色执行状态响应模型（对齐 openapi.yaml WorkerTaskOut）。"""

from datetime import datetime

from app.schemas import StrictModel


class WorkerTaskOut(StrictModel):
    """角色任务状态项。"""

    id: int
    worker_role: str
    task_status: str
    stage_name: str | None = None
    started_at: datetime | None = None
    finished_at: datetime | None = None


class WorkerListData(StrictModel):
    """角色任务列表数据。"""

    list: list[WorkerTaskOut]

File: code-sec-evaluator/frontend/src/types/worker.ts
import type { TaskStatus, WorkerRole } from './enums';

export interface WorkerTaskItem {
  id: number;
  worker_role: WorkerRole;
  task_status: TaskStatus;
  stage_name: string | null;
  started_at: string | null;
  finished_at: string | null;
}
